src/test/java/testsuite/simple/AwsIamAuthenticationHelperTest.java

```diff
@@ -158,4 +158,15 @@ public void test_6_InvalidRegion() {
             )
         );
     }
+    /**
+     *
+     */
+    @Test
+    public void test_7_ValidChinaHostAndRegion() {
+        Assertions.assertNotNull(new AwsIamAuthenticationTokenHelper(
```

Comment:
maybe you should add a couple of examples that have the `.cn` at the end but have other irregularities in the middle of the URL to be sure your regex is catching what you want